tst/hashmap.tst

```diff
@@ -375,7 +375,8 @@ gap> hashmap["foo"];
 fail
 
 # now insert a key to which we keep a reference
-gap> foo:="foo";;
+# must be immutable else an immutable copy will be made on insertion
+gap> foo:=Immutable("foo");;
```

Comment:
Huhm, that's a bummer. I really would like to have hash maps and sets that use object identity for their keys, similar to OBJ_SET and OBJ_MAP in GAP itself, and also `IdDict` in Julia. But I guess it might be better (also from a performance perspective?) to implement those with some additional C code anyway. So, whatever...